application/common/models/Reset.php

```diff
@@ -487,23 +484,23 @@ public function setType($type, $methodId = null)
         if (in_array($type, [self::TYPE_SPOUSE, self::TYPE_SUPERVISOR, self::TYPE_PRIMARY])) {
             $this->type = $type;
             $this->method_id = null;
-        } elseif ($type == self::TYPE_METHOD) {
+        } elseif (in_array($type, [self::TYPE_METHOD, Method::TYPE_EMAIL, Method::TYPE_PHONE])) {
```

Comment:
Yeah it might seem a bit weird. Basically we have these "meta" types of supervisor, spouse, and primary, but then they can also be the preconfigured options and when one of the preconfigured options we also need their unique id.